### PR TITLE
.Net: Some code enhancements

### PR DESCRIPTION
**Pull request by WeihanLi**
### Motivation and Context

To keep the code clean and follow the best practice

### Description

- add `readonly` for field if applicable
- sealed internal types if applicable
- prefer static method other than instance method if applicable
- remove unnecessary init
- prefer smaller access scope
- fix two typo

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Plugins/Plugins.Web/Bing/BingConnector.cs [hunk 1]
```diff
@@ -53,9 +53,7 @@ public BingConnector(string apiKey, HttpClient httpClient, ILoggerFactory? logge
     /// <inheritdoc/>
     public async Task<IEnumerable<string>> SearchAsync(string query, int count = 1, int offset = 0, CancellationToken cancellationToken = default)
     {
-        if (count <= 0) { throw new ArgumentOutOfRangeException(nameof(count)); }
-
-        if (count >= 50) { throw new ArgumentOutOfRangeException(nameof(count), $"{nameof(count)} value must be less than 50."); }
+        if (count is <= 0 or >= 50) { throw new ArgumentOutOfRangeException(nameof(count), count, $"{nameof(count)} value must be less than 50."); }
```
[the remark this answers is not in the document]
Comment (WeihanLi) [reply dotnet/src/Plugins/Plugins.Web/Bing/BingConnector.cs]: @dmytrostruk could you please help review again, thanks very much

**Comment by WeihanLi** [comment]
For `CA1805`, currently, it's disabled, is there any reason for it, should we enable?

https://github.com/microsoft/semantic-kernel/blob/76db027273371ea81e6db66afcb1d888cc53b459/.editorconfig#L163

**Review by dmytrostruk — APPROVED** [review]

File: dotnet/src/Plugins/Plugins.Web/Bing/BingConnector.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmytrostruk) [reply dotnet/src/Plugins/Plugins.Web/Bing/BingConnector.cs]: Looks good to me, thanks!

**Comment by dmytrostruk** [comment]
> For `CA1805`, currently, it's disabled, is there any reason, should we enable it? And for other disabled rules?

I would suggest waiting with other rules, as we will need to communicate that with internal team and decide which rules we want to enable in the future. 
If you see some rules, which in your opinion can be enabled to improve the code, feel free to create an issue with your proposal and what benefits it will bring to current codebase. Then, we will review the proposal and respond with decision. 

In this case, your PR review will be faster, as we will be already in the context why specific rule was enabled/disabled.

**Review by SergeyMenshykh — APPROVED** [review]

**Pull request merged**